inc/namespace.php

```diff
@@ -1006,128 +1018,129 @@ function elasticpress_mapping( array $mapping ) : array {
+			}
+			// Check full field path against `path_unmatch` to exclude patterns after matching.
+			if ( $template['path_unmatch'] && fnmatch( $template['path_unmatch'], $field ) ) {
+				continue;
 			}
 
-			// This is a path mapping.
-			if ( count( $sub_fields ) > 1 ) {
-				// Remove the first part of the path as it is handled below.
-				array_shift( $sub_fields );
-				$sub_fields = array_reverse( $sub_fields );
-				$field_mapping = array_reduce( $sub_fields, function ( $carry, $field ) {
-					return [ 'properties' => [ $field => $carry ] ];
-				}, $field_mapping );
+			// We have a matching template so update the field mapping for a text field or object property.
+			if ( ( $template['mapping']['type'] ?? false ) === 'text' ) {
+				$field_mapping = array_merge_recursive_distinct( $template['mapping'], $field_mapping );
+			}
+			if ( $template['mapping']['properties'][ $end_field ] ?? false ) {
+				$template['mapping']['properties'][ $end_field ] = array_merge_recursive_distinct(
+					$template['mapping']['properties'][ $end_field ],
+					$field_mapping
+				);
+				$field_mapping = [ 'properties' => $template['mapping']['properties'] ];
+				// Remove end field from $sub_fields if matched.
+				array_pop( $sub_fields );
 			}
+		}
 
-			// Merge mapping.
-			$mapping['mappings'][ $type ]['properties'] = array_merge_recursive_distinct(
-				$mapping['mappings'][ $type ]['properties'],
-				[ $start_field => $field_mapping ]
-			);
+		// This is a path mapping.
+		if ( count( $sub_fields ) > 1 ) {
+			// Remove the first part of the path as it is handled below.
+			array_shift( $sub_fields );
+			$sub_fields = array_reverse( $sub_fields );
+			$field_mapping = array_reduce( $sub_fields, function ( $carry, $field ) {
+				return [ 'properties' => [ $field => $carry ] ];
+			}, $field_mapping );
 		}
+
+		// Merge mapping.
+		$new_mapping['properties'] = array_merge_recursive_distinct(
+			$new_mapping['properties'],
+			[ $start_field => $field_mapping ]
+		);
 	}
 
 	// Ensure text fields with a field name have the `.sortable` field.
 	// Some ES version mappings are missing this.
-	foreach ( $mapping['mappings'] as $type => $map ) {
-		if ( ! isset( $map['properties'] ) ) {
-			continue;
-		}
-		foreach ( $map['properties'] as $name => $property ) {
-			if ( $property['type'] === 'text' && isset( $property['fields'] ) ) {
-				$mapping['mappings'][ $type ]['properties'][ $name ]['fields'] = array_merge_recursive_distinct( $property['fields'], [
-					'sortable' => [
-						'type'         => 'keyword',
-						'ignore_above' => 10922,
-						'normalizer'   => 'lowerasciinormalizer',
-					],
-				] );
-			}
+	foreach ( $new_mapping['properties'] as $name => $property ) {
+		if ( $property['type'] === 'text' && isset( $property['fields'] ) ) {
+			$new_mapping['properties'][ $name ]['fields'] = array_merge_recursive_distinct( $property['fields'], [
+				'sortable' => [
+					'type'         => 'keyword',
+					'ignore_above' => 10922,
```

Comment:
Could we extract this magic number into a descriptive named constant?

Reply:
It's just matching the default ElasticPress mapping for sortable fields. We could have a constant but it'd only be used this once.

Reply:
Yeah, I would still move to a constant